packages/block-library/src/template-part/edit/inner-blocks.js

```diff
@@ -127,11 +127,14 @@ export default function TemplatePartInnerBlocks( {
 	const { canViewTemplatePart, canEditTemplatePart } = useSelect(
 		( select ) => {
 			return {
-				canViewTemplatePart:
-					select( coreStore ).canUser( 'read', 'templates' ) ?? false,
-				canEditTemplatePart:
-					select( coreStore ).canUser( 'create', 'templates' ) ??
-					false,
+				canViewTemplatePart: !! select( coreStore ).canUser( 'read', {
+					kind: 'postType',
+					name: 'wp_template',
+				} ),
+				canEditTemplatePart: !! select( coreStore ).canUser( 'create', {
```

Comment:
Not from this PR, but it felt odd that we call this "can edit" but check for whether they can create (`POST`) and not `update` (PUT). Haven't checked if that makes more sense in the context it's used.

Reply:
I think introducing the pattern might have been my fault. The WP maps all template [CRUD permissions to the same capability](https://github.com/WordPress/wordpress-develop/blob/5b0006ca9b5408aa73b6dc71fb5b3ebf85f497ec/src/wp-includes/post.php#L387-L398), so I reused already existing call for edit capability check to avoid making extra `OPTIONS` request to the server.

The duplicate requests issue was fixed (#43480). I forgot to update my old code, and the pattern spread into new places.

I will follow up on this in a separate PR.